core/modules/node/node.module

```diff
@@ -2449,7 +2449,12 @@ function node_form_search_form_alter(&$form, $form_state) {
     );
 
     // Node types:
-    $types = array_map('check_plain', node_type_get_names());
+    $types = array();
+    foreach (node_type_get_types() as $type => $detail) {
+      if (!$detail->settings['hidden_path'] || user_access('view hidden paths') ) {
+        $types[$type] = $detail->name;
```

Comment:
Somehow "check_plain" was dropped. But node type names are provided by users/admins and they could contain markup (gets saved to the database). I think, $detail->name should get wrapped in check_plain(), so it is safe to use it in the options.